src/array_api_stubs/_draft/_types.py

```diff
@@ -51,6 +56,65 @@
 # to keep pyflakes happy. https://github.com/python/typeshed/issues/3556
 ellipsis = TypeVar("ellipsis")
 
+DefaultDataTypes = TypedDict(
+    "DefaultDataTypes",
+    {
+        "real floating": dtype,
+        "complex floating": dtype,
+        "integral": dtype,
+        "indexing": dtype,
+    },
+)
+DataTypes = TypedDict(
+    "DataTypes",
+    {
+        "bool": dtype,
+        "float32": dtype,
+        "float64": dtype,
+        "complex64": dtype,
+        "complex128": dtype,
+        "int8": dtype,
+        "int16": dtype,
+        "int32": dtype,
+        "int64": dtype,
+        "uint8": dtype,
+        "uint16": dtype,
+        "uint32": dtype,
+        "uint64": dtype,
+    },
+    total=False,
+)
+Capabilities = TypedDict(
+    "Capabilities", {"boolean_indexing": bool, "data_dependent_shapes": bool}
+)
+
+
+@dataclass
+class Info:
```

Comment:
Small note mostly relevant for downstream usage: dataclasses very much follow a nominal subtyping pattern (as opposed to protocols), so the only way downstream `Info` implementations can properly subtype `Info` is by directly inheriting from this very specific class here. Assuming this is undesired, might it be an idea to type it as a protocol instead?

Also what's going on with all the `@staticmethod` decorators? Is this something that's necessary due to some dataclass-specific reasons?

Reply:
@BvB93 No, no dataclass-specific reasons. Do you have recommendations for how to type returning a namespace? Specifically, this `Info` class is intended to document the return value for `__array_namespace_info__()`. If you have recommendations for the right way to type this, LMK.

Reply:
In this instance, my intent was to type an object with various static methods.

Reply:
It's quite possible that typing as a protocol makes more sense, as proposed in https://github.com/data-apis/array-api/pull/685.

Reply:
Yeah, I'd definitely recommend protocol usage here as, in its most basic forms, those are used to denote "an arbitrary object with X attributes and Y methods", so they're fully duck-typable.

Reply:
Refactored using protocols, so I'll mark this as resolved.